src/platform/platform_misc.c

```diff
@@ -68,29 +69,51 @@ static int windows_system(const char *cmd)
 {
     PROCESS_INFORMATION p_info;
     STARTUPINFOW s_info;
-    DWORD ReturnValue;
+    HANDLE hp, p_stderr;
+    DWORD flags, ret, err = ERROR_SUCCESS;
 
     memset(&s_info, 0, sizeof(s_info));
     memset(&p_info, 0, sizeof(p_info));
     s_info.cb = sizeof(s_info);
+    s_info.dwFlags = STARTF_USESTDHANDLES;
 
-    char_os *utf16cmd;
-    utf16cmd = caml_stat_strdup_to_os(cmd);
-    if (CreateProcessW(NULL, utf16cmd, NULL, NULL, FALSE, CREATE_NO_WINDOW, NULL, NULL, &s_info, &p_info))
-    {
+    /* Duplicate stderr to make sure it is inheritable */
+    p_stderr = GetStdHandle(STD_ERROR_HANDLE);
+    if (p_stderr == INVALID_HANDLE_VALUE) {
+        err = GetLastError(); goto ret1;
+    }
+    hp = GetCurrentProcess();
+    if (! DuplicateHandle(hp, p_stderr, hp, &(s_info.hStdError),
+                          0, TRUE, DUPLICATE_SAME_ACCESS)) {
+        err = GetLastError(); goto ret1;
+    }
```

Comment:
I don't understand why all this is necessary. I'm not very familiar with all this "inheritable" intrincacies, but if we are setting stderr to stdout, shouldn't we make sure that `stdout` is inheritable (which we already do in the call to `Duplicate Handle below)?